src/Weather_forecast_on_dashboard.py

```diff
@@ -0,0 +1,133 @@
+
+@st.cache_data(show_spinner=False)
+def fetch_data_for_init_time(init_time, forecast_date, lat, lon, parameter, model="ifs"):
+    # Adjust the initialization datetime based on the init_time offset in hours
+    init_datetime = forecast_date - timedelta(hours=init_time)  # This will be a datetime object
+
+    FH = FastHerbie([init_datetime], model=model, fxx=range(0, 24, 1), fast=True)
+
+    try:
+        FH.download()  # Ensure the file is downloaded
+    except Exception as e:
+        st.error(f"Failed to download data for initialization time {init_datetime}. Error: {e}")
+        return None, None
+
+    variable_subset = ":10[u|v]" if parameter == "u10:v10" else ":100[u|v]" if parameter == "u100:v100" else "2t"
+
+    try:
+        ds = FH.xarray(variable_subset, remove_grib=False)
+        if isinstance(ds, list):
+            ds = ds[0]
+    except EOFError as e:
+        st.error(f"Failed to read data file for {init_datetime}. Error: {e}")
+        return None, None
+    except Exception as e:
+        st.error(f"An unexpected error occurred while reading data: {e}")
+        return None, None
+
+    return ds, init_datetime
+
+def process_initialization(init_time, forecast_date, lat, lon, parameter, model="ifs"):
+    ds, init_datetime = fetch_data_for_init_time(init_time, forecast_date, lat, lon, parameter, model)
+    if ds is None:
+        return []
+
+    interpolated_data = []
+
+    time_dim = ds['step'] if 'step' in ds.dims else ds['time'] if 'time' in ds.dims else None
+    if time_dim is None:
+        st.error("Neither 'step' nor 'time' dimension is present in the dataset. Unable to proceed with forecast.")
+        return []
+
+    for time_val in time_dim:
+        actual_forecast_time = init_datetime + pd.to_timedelta(time_val.values)
+        if actual_forecast_time.date() == forecast_date.date():
+            if parameter == "u10:v10":
+                u = ds['u10'].interp(latitude=lat, longitude=lon, method="nearest", step=time_val)
+                v = ds['v10'].interp(latitude=lat, longitude=lon, method="nearest", step=time_val)
+                value = np.sqrt(u**2 + v**2)
+            elif parameter == "u100:v100":
+                u = ds['u100'].interp(latitude=lat, longitude=lon, method="nearest", step=time_val)
+                v = ds['v100'].interp(latitude=lat, longitude=lon, method="nearest", step=time_val)
+                value = np.sqrt(u**2 + v**2)
+            else:  # temp
+                value = ds['t2m'].interp(latitude=lat, longitude=lon, method="nearest", step=time_val) - 273.15
+
+            interpolated_data.append({
+                'date_time': actual_forecast_time,
+                'latitude': lat,
+                'longitude': lon,
+                'value': value.compute().item(),
+                'init_time': f"{init_time} hours before"
+            })
+
+    return interpolated_data
+
+def get_forecast(forecast_date, lat, lon, parameter, init_times, model="ifs"):
```

Comment:
try to add type hinting too https://docs.python.org/3/library/typing.html, it make it more readable

forecast_date:dateimte, lat:float, lon:float, ......